tests/build_/test_spec.py

```diff
@@ -117,3 +117,58 @@ def test_image_tag(self, rand, timestr, platform, arrangement, has_platform_suff
             img_tag += '-{platform}'
         img_tag = img_tag.format(random_number=rand, time_string=timestr, **kwargs)
         assert spec.image_tag.value == img_tag
+
+    @pytest.mark.parametrize(('odcs_enabled', 'signing_intent', 'compose_ids',
+                              'yum_repourls', 'exc'), (
+        (False, 'release', [1, 2], ['http://example.com/my.repo'], OsbsValidationException),
+        (False, 'release', [1, 2], None, OsbsValidationException),
+        (False, None, [1, 2], ['http://example.com/my.repo'], OsbsValidationException),
+        (False, 'release', None, ['http://example.com/my.repo'], OsbsValidationException),
+        (False, 'release', None, None, OsbsValidationException),
+        (False, None, [1, 2], None, OsbsValidationException),
+        (False, None, None, ['http://example.com/my.repo'], None),
+        (False, None, None, None, None),
+        (True, 'release', [1, 2], ['http://example.com/my.repo'], OsbsValidationException),
+        (True, 'release', [1, 2], None, OsbsValidationException),
+        (True, None, [1, 2], ['http://example.com/my.repo'], OsbsValidationException),
+        (True, 'release', None, ['http://example.com/my.repo'], None),
+        (True, 'release', None, None, None),
+        (True, None, [1, 2], None, None),
+        (True, None, "string", None, OsbsValidationException),
```

Comment:
A good test case here would be that compose_ids is a int instead of a list of ints.

Reply:
right will add that as well